### PR TITLE
Fix accordion stack purposes

### PR DESCRIPTION
**Pull request by allisonking**
Closes https://github.com/ethyca/fides/issues/4257

### Description Of Changes

Prevents a stack that only has special features from showing up with empty purposes.

| Before | After |
|--------|--------|
| image  | image |

### Code Changes

* [x] it was easy, just needed to make sure the `purposes` had a length
* [x] while I was here, I also sorted the purposes (see Dawn's catch here: https://ethyca.slack.com/archives/C05E74TR0BB/p1697215553763889)

### Steps to Confirm

* Go to the admin ui and for a data use that shows up in the TCF overlay, add two special features: `Use precise geolocation data` and `Actively scan device characteristics for identification`
* These will combine in the banner stack to "Precise geolocation data, and identification through device scanning". Clicking on it should not show an empty "Purposes" block

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* [x] Issue Requirements are Met
* [x] Update `CHANGELOG.md`

### CODE REVIEW

File: clients/fides-js/src/lib/tcf.ts [hunk 1]
```diff
@@ -47,7 +47,7 @@ const generateAcString = ({
         .map((id) => decodeVendorId(id).id)
     )
   );
-  const vendorIds = uniqueIds.sort().join(".");
+  const vendorIds = uniqueIds.sort((a, b) => Number(a) - Number(b)).join(".");
```
Comment (allisonking) [inline clients/fides-js/src/lib/tcf.ts:50]: it turns out JS sorts alphabetically by default, even for numbers. this means `.sort()` doesn't always work, i.e. it'll put `10` before `9`! but if we explicitly give it a sort function, it'll figure itself out

**Comment by galvana** [comment]
Looks good to me
toggle

**Review by galvana — APPROVED** [review]
Tested locally and it works as expected!

**Comment by allisonking** [comment]
Failing test is a known issue: https://github.com/ethyca/fides/issues/4288

**Pull request merged**